subsys/zigbee/cli/zigbee_cli_cmd_zdo.c

```diff
@@ -0,0 +1,2228 @@
+		ret_err = -EINVAL;
+		goto error;
+	}
+
+	/* Make sure ZBOSS buffer API is called safely. */
+	zb_osif_disable_all_inter();
+	bufid = zb_buf_get_out();
+	zb_osif_enable_all_inter();
+
+	if (!bufid) {
+		zb_cli_print_error(shell, "Failed to execute command (buf alloc failed)",
+				   ZB_FALSE);
+		ret_err = -ENOEXEC;
+		goto error;
+	}
+
+	bind_req = ZB_BUF_GET_PARAM(bufid, zb_zdo_mgmt_bind_param_t);
+	ZB_BZERO(bind_req, sizeof(*bind_req));
+	bind_req->start_index = tsn_ctx_entry->cmd_ctx.req_seq.start_index;
+	bind_req->dst_addr = tsn_ctx_entry->cmd_ctx.req_seq.dst_addr;
+
+	/* Initialize context and send a request. */
+	tsn_ctx_entry->shell = shell;
+	tsn_ctx_entry->zdo_req.buffer_id = bufid;
+	tsn_ctx_entry->zdo_req.req_fn = zb_zdo_mgmt_bind_req;
+	tsn_ctx_entry->zdo_req.req_cb_fn = cmd_zb_mgmt_bind_cb;
+	tsn_ctx_entry->zdo_req.ctx_timeout = 0;
+	tsn_ctx_entry->zdo_req.timeout_cb_fn = NULL;
+
+	zb_err_code = ZB_SCHEDULE_APP_CALLBACK(zb_zdo_req,
+					       (tsn_ctx_entry - tsn_ctx));
+	if (zb_err_code != RET_OK) {
+		zb_cli_print_error(shell, "Unable to schedule zdo request",
+				   ZB_FALSE);
+		ret_err = -ENOEXEC;
+		goto error;
+	}
+
+	return ret_err;
+
+error:
+	if (bufid != 0) {
+		/* Make sure ZBOSS buffer API is called safely. */
+		zb_osif_disable_all_inter();
+		zb_buf_free(bufid);
+		zb_osif_enable_all_inter();
+	}
+	if (tsn_ctx_entry != NULL) {
+		invalidate_ctx(tsn_ctx_entry);
+	}
+	return ret_err;
+}
+
+/**@brief Callback for a single mgmt_lqi_req transaction
+ *
+ * @note
+ * When the lqi table is too large to fit into a single mgmt_bind_rsp command
+ * frame, this function will issue a new mgmt_lqi_req to download reminder of
+ * the table. This process may involve several round trips of mgmt_lqi_req
+ * followed by mgmt_lqi_resp until the whole binding table is downloaded.
+ *
+ * @param bufid     Reference to ZBOSS buffer (as required by Zigbee stack API)
+ */
+static bool zdo_mgmt_lqi_cb(struct zdo_tsn_ctx *tsn_ctx_entry, zb_bufid_t bufid)
+{
+	zb_zdo_neighbor_table_record_t *tbl_rec;
+	zb_zdo_mgmt_lqi_resp_t *resp;
+	bool result = true;
+
+	resp = (zb_zdo_mgmt_lqi_resp_t *)zb_buf_begin(bufid);
+
+	if (resp->status == ZB_ZDP_STATUS_SUCCESS) {
+		if (resp->start_index ==
+		    tsn_ctx_entry->cmd_ctx.req_seq.start_index) {
+			shell_print(tsn_ctx_entry->shell,
+				    "[idx] ext_pan_id       ext_addr         short_addr flags permit_join depth lqi");
+		}
+
+		tbl_rec = (zb_zdo_neighbor_table_record_t *)((uint8_t *)resp +
+							     sizeof(*resp));
+
+		for (uint8_t i = 0; i < resp->neighbor_table_list_count; i++) {
+			shell_fprintf(tsn_ctx_entry->shell, SHELL_NORMAL,
+				      "[%3u] ", resp->start_index + i);
+
+			zb_cli_print_eui64(tsn_ctx_entry->shell,
+					   tbl_rec->ext_pan_id);
+
+			shell_fprintf(tsn_ctx_entry->shell, SHELL_NORMAL, " ");
+			zb_cli_print_eui64(tsn_ctx_entry->shell,
+					   tbl_rec->ext_addr);
+
+			shell_print(tsn_ctx_entry->shell, " 0x%04x     0x%02x  %u           %u     %u",
```

Comment:
You can use variable padding with `*` here. shell supports the same syntax as `printf` function.

```
int main(void)                                                                                                           
{                                                                                                                        
        int i = 10;                                                                                                      
        int padding = 4;                                                                                                
                                                                                                                         
        printf("test:%*u test\n", padding, i);                                                                          
                                                                                                                         
        return 0;                                                                                                        
}  
```
will result in:

`test  10 test`  // test__10_test

Reply:
Well in fact you do not need variable padding :) Regular one would be enough and you will save some flash.